lib/xcpretty/syntax.rb

```diff
@@ -1,44 +1,39 @@
+begin
+  require 'rouge'
+rescue LoadError
+  Rouge = nil
+end
+
 require 'xcpretty/snippet'
 
 module XCPretty
   class Syntax
-    def self.register_filetype(type, extensions)
-      @filetypes ||= {}
-      extensions.each { |ext| @filetypes[ext] = type }
-    end
-
-    register_filetype 'c++',    ['.cpp', '.hpp', '.c++', '.cxx', '.cc']
-    register_filetype 'objc',   ['.m', '.h']
-    register_filetype 'objc++', ['.mm', '.hh']
-    register_filetype 'swift',  ['.swift']
-    register_filetype 'dylan',  ['.dyl', '.dylan']
-    register_filetype 'ruby',   ['.ruby', '.rb']
+    def self.highlight(snippet)
+      return snippet.contents unless Rouge
 
-    def self.highlight(snippet, options = '')
-      if Pygments.available?
-        language = file_language(File.basename(snippet.file_path))
-        Pygments.pygmentize(snippet.contents, language, options)
+      if snippet.file_path.include?(':')
+        filename = snippet.file_path.rpartition(':').first
       else
-        snippet.contents
+        filename = snippet.file_path
       end
-    end
-
-    private
 
-    def self.file_language(filename)
-      ext = File.extname(filename)
-      @filetypes[ext] || 'objc'
-    end
-  end
+      options = {
+        :filename => File.basename(filename),
+        :source => snippet.contents,
+      }
+      lexer = Rouge::Lexer.guesses(options).first
```

Comment:
I think both should be here, and here's a few examples of why:
- This code using the Quick/Nimble Swift framework isn't able to be detected as Swift, purely because it would also be valid syntax in other languages such as Ruby and Python. The file extension is required to improve the guessing.
    ![screen shot 2015-06-25 at 16 01 49](https://cloud.githubusercontent.com/assets/44164/8367497/93f856fe-1b53-11e5-901f-0793c0713442.png)
- Filename isn't enough to match Objective-C, since `.m` is a file extension shared with other file types (matlab for one).